lib/features/auth/presentation/signup/view_model/signup_cubit.dart

```diff
@@ -0,0 +1,78 @@
+import 'package:bloc/bloc.dart';
+import 'package:flowery_e_commerce/features/auth/data/model/response/signup_response_dto.dart';
+import 'package:flowery_e_commerce/features/auth/domain/usecase/auth_usecase.dart';
+import 'package:flutter/material.dart';
+import 'package:injectable/injectable.dart';
+import 'package:meta/meta.dart';
+import '../../../../../core/networking/common/api_result.dart';
+import '../../../domain/entities/signup_entity.dart';
+
+part 'signup_state.dart';
+@injectable
+class SignupCubit extends Cubit<SignupState> {
+  SignupCubit(this._authUseCase) : super(SignupInitial());
+  final AuthUseCase _authUseCase;
```

Comment:
its not best practice to declare  TextEditingController

Reply:
i suggest refactoring this using Flutter hooks package 
i preferer to vote on this step or refactor it with best practice for declaration